services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,192 @@
+        }
+        if (maxBufferSize != null ? !maxBufferSize.equals(that.maxBufferSize) : that.maxBufferSize != null) {
+            return false;
+        }
+        return maxBatchOpenInMs != null ? maxBatchOpenInMs.equals(that.maxBatchOpenInMs) : that.maxBatchOpenInMs == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = maxBatchItems != null ? maxBatchItems.hashCode() : 0;
+        result = 31 * result + (maxBatchKeys != null ? maxBatchKeys.hashCode() : 0);
+        result = 31 * result + (maxBufferSize != null ? maxBufferSize.hashCode() : 0);
+        result = 31 * result + (maxBatchOpenInMs != null ? maxBatchOpenInMs.hashCode() : 0);
+        return result;
+    }
+
+    public static final class Builder implements CopyableBuilder<Builder, BatchOverrideConfiguration> {
+
+        private Integer maxBatchItems;
+        private Integer maxBatchKeys;
+        private Integer maxBufferSize;
+        private Duration maxBatchOpenInMs;
+
+        private Builder() {
+        }
+
+        /**
+         * Define the the maximum number of messages that are batched together in a single request.
+         *
+         * @param maxBatchItems The new maxBatchItems value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBatchItems(Integer maxBatchItems) {
+            this.maxBatchItems = maxBatchItems;
+            return this;
+        }
+
+        /**
+         * Define the maximum number of batchKeys to keep track of. A batchKey determines which requests are batched together
+         * and is calculated by the client based on the information in a request.
+         * <p>
+         * Ex. SQS determines a batchKey based on a request's queueUrl in combination with its overrideConfiguration, so
+         * requests with the same queueUrl and overrideConfiguration will have the same batchKey and be batched together.
+         *
+         * @param maxBatchKeys the new maxBatchKeys value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBatchKeys(Integer maxBatchKeys) {
+            this.maxBatchKeys = maxBatchKeys;
+            return this;
+        }
+
+        /**
+         * Define the maximum number of items to allow to be buffered for each batchKey.
+         *
+         * @param maxBufferSize the new maxBufferSize value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBufferSize(Integer maxBufferSize) {
+            this.maxBufferSize = maxBufferSize;
+            return this;
+        }
+
+        /**
+         * The maximum amount of time (in milliseconds) that an outgoing call waits for other requests before sending out a batch
+         * request.
+         *
+         * @param maxBatchOpenInMs The new maxBatchOpenInMs value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBatchOpenInMs(Duration maxBatchOpenInMs) {
```

Comment:
The most common term to denote milliseconds is `Millis`. Can we change?

Reply:
Or just remove `InMs` since it's a Duration ;) We should probably discuss naming in API surface area review because I'm not sure what maxBatchOpen means...